### ISSUE TITLE
Getting DDL load failed Errors when trying to import Open3D

### ISSUE THREAD

**Issue by Me817** [intel-isl/Open3D#2136]
Hello,

**Describe the bug**
I have installed Open3D 0.10.0.1 in an virtual environment on my Windows 10 Desktop PC, but every time when I try to import open3d module I get the following error message:

```
Traceback (most recent call last):
  File "C:/Users/PC/PycharmProjects/bachelorthesis/interactive_visualization.py", line 9, in <module>
    import open3d as o3d
  File "C:\Users\PC\AppData\Local\Programs\Python\Python37\lib\site-packages\open3d\__init__.py", line 47, in <module>
    from open3d.open3d_pybind import camera
ImportError: DLL load failed: Das angegebene Modul wurde nicht gefunden.
```
I also get this error message when I try to run the python examples.

**To Reproduce**
To reproduce run any python example where you have an open3d import.

**Expected behavior**
I expected that all examples and the import works without any error.

**Environment (please complete the following information):**

- Operating system: Windows 10 64-bit
- Python version: Python 3.7.4
- Open3D version: 0.10.0.1
- Is this remote workstation?: no
- How did you install Open3D?: pip
- Compiler version (if built from source): -/-

**Additional context**
I installed in virtual environment venv in the following path: C:\Users\PC\Bachelorthesis\venv\Lib\site-packages\open3d. And I cloned it from github via GitBash into C:\Users\PC.

Is there anything I can do to make it run without errors? I already tried the steps mentioned in #1872, but it did not work for me.
Thank you in advance.

**Comment by yxlao** [comment]
Hi @Me817 , let's try to debug this issue. First, download the "Dependencies" app as mentioned in:
https://vxlabs.com/2017/12/06/how-to-debug-pyinstaller-dll-pyd-load-failed-issues-on-windows/#seeing-the-light-with-lucasg8217s-dependencies

Using the "Dependencies" tool, locate the `C:\Users\PC\AppData\Local\Programs\Python\Python37\lib\site-packages\open3d` directory, and locate the python shared library named similar to `open3d_pybind.cp35-win_amd64.pyd`. The app will tell us which DLL is missing. Please provide a screenshot showing the missing DLL. Thanks.

**Comment by collikoira** [comment]
![open3d](https://user-images.githubusercontent.com/32222463/89706430-4544fb00-d96e-11ea-8f0c-0e589506051d.png)
I had the same issue, took @yxlao advise and seems that python36.dll is not loaded

**Comment by yxlao** [comment]
The missing `python36.dll` is likely due virtualenv settings or python runtime version mismatch (e.g. the python env where you install Open3D is not the python env you run Open3D).

@collikoira could you do the following steps?
1) locate the directory of the `python.exe` within the virtualenv that you have installed Open3D (from your screenshot looks like it's somewhere under `C:\Users\OMISTAJA\PycharmProjects\pythonProject\venv`)
2) open a terminal, `cd` to that directory, run `.\python.exe` and make sure it says python 3.6 and 64 bit.
3) in the same directory, see if `python36.dll` is there. In my case, `python36.dll` is located at the same folder as `python.exe`. 
4) in the same directory, run `.\python.exe -c "import open3d"` see if it still complains about missing DLL

**Comment by Me817** [comment]
HI @yxlao , I tried the steps you mentioned above. But the `open3d_pybind.cp37-win_amd64.pyd` is not in `C:\Users\PC\AppData\Local\Programs\Python\Python37\lib\site-packages\open3d ` and not in the virtual environment (`C:\Users\PC\Bachelorthesis\venv\Lib\site-packages\open3d`). What can I do in this case?

Here you can see how I tried to install open3d:
```
Microsoft Windows [Version 10.0.19041.388]
(c) 2020 Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\PC>python --version
Python 3.7.4

C:\Users\PC>pip --version
pip 19.0.3 from c:\users\pc\appdata\local\programs\python\python37\lib\site-packages\pip (python 3.7)

C:\Users\PC>python -m pip install --upgrade pip==20.2
Collecting pip==20.2
  Using cached https://files.pythonhosted.org/packages/36/74/38c2410d688ac7b48afa07d413674afc1f903c1c1f854de51dc8eb2367a5/pip-20.2-py2.py3-none-any.whl
Installing collected packages: pip
  Found existing installation: pip 19.0.3
    Uninstalling pip-19.0.3:
      Successfully uninstalled pip-19.0.3
Successfully installed pip-20.2

C:\Users\PC>pip install virtualenv==20.0.21
Collecting virtualenv==20.0.21
  Using cached virtualenv-20.0.21-py2.py3-none-any.whl (4.7 MB)
Collecting six<2,>=1.9.0
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting importlib-metadata<2,>=0.12; python_version < "3.8"
  Using cached importlib_metadata-1.7.0-py2.py3-none-any.whl (31 kB)
Collecting filelock<4,>=3.0.0
  Using cached filelock-3.0.12-py3-none-any.whl (7.6 kB)
Collecting appdirs<2,>=1.4.3
  Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting distlib<1,>=0.3.0
  Using cached distlib-0.3.1-py2.py3-none-any.whl (335 kB)
Collecting zipp>=0.5
  Using cached zipp-3.1.0-py3-none-any.whl (4.9 kB)
Installing collected packages: six, zipp, importlib-metadata, filelock, appdirs, distlib, virtualenv
Successfully installed appdirs-1.4.4 distlib-0.3.1 filelock-3.0.12 importlib-metadata-1.7.0 six-1.15.0 virtualenv-20.0.21 zipp-3.1.0
WARNING: You are using pip version 20.2; however, version 20.2.1 is available.
You should consider upgrading via the 'c:\users\pc\appdata\local\programs\python\python37\python.exe -m pip install --upgrade pip' command.

C:\Users\PC>virtualenv --version
virtualenv 20.0.21 from c:\users\pc\appdata\local\programs\python\python37\lib\site-packages\virtualenv\__init__.py

C:\Users\PC>mkdir Bachelorthesis

C:\Users\PC>cd Bachelorthesis

C:\Users\PC\Bachelorthesis>virtualenv venv
created virtual environment CPython3.7.4.final.0-64 in 517ms
  creator CPython3Windows(dest=C:\Users\PC\Bachelorthesis\venv, clear=False, global=False)
  seeder FromAppData(download=False, pip=latest, setuptools=latest, wheel=latest, via=copy, app_data_dir=C:\Users\PC\AppData\Local\pypa\virtualenv\seed-app-data\v1.0.1)
  activators BashActivator,BatchActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

C:\Users\PC\Bachelorthesis>venv\Scripts\activate

(venv) C:\Users\PC\Bachelorthesis>pip --version
pip 20.1.1 from C:\Users\PC\Bachelorthesis\venv\lib\site-packages\pip (python 3.7)

(venv) C:\Users\PC\Bachelorthesis>python -m pip install --upgrade pip==20.2
Collecting pip==20.2
  Using cached pip-20.2-py2.py3-none-any.whl (1.5 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.1.1
    Uninstalling pip-20.1.1:
      Successfully uninstalled pip-20.1.1
Successfully installed pip-20.2

(venv) C:\Users\PC\Bachelorthesis>pip install requests
Collecting requests
  Using cached requests-2.24.0-py2.py3-none-any.whl (61 kB)
Collecting idna<3,>=2.5
  Using cached idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting chardet<4,>=3.0.2
  Using cached chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1
  Using cached urllib3-1.25.10-py2.py3-none-any.whl (127 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2020.6.20-py2.py3-none-any.whl (156 kB)
Installing collected packages: idna, chardet, urllib3, certifi, requests
Successfully installed certifi-2020.6.20 chardet-3.0.4 idna-2.10 requests-2.24.0 urllib3-1.25.10
WARNING: You are using pip version 20.2; however, version 20.2.1 is available.
You should consider upgrading via the 'c:\users\pc\bachelorthesis\venv\scripts\python.exe -m pip install --upgrade pip' command.

(venv) C:\Users\PC\Bachelorthesis>pip install open3d
Collecting open3d
  Using cached open3d-0.10.0.1-cp37-cp37m-win_amd64.whl (4.4 MB)
Collecting notebook
  Using cached notebook-6.1.1-py3-none-any.whl (9.4 MB)
Collecting matplotlib
  Using cached matplotlib-3.3.0-cp37-cp37m-win_amd64.whl (8.8 MB)
Collecting numpy
  Using cached numpy-1.19.1-cp37-cp37m-win_amd64.whl (12.9 MB)
Collecting widgetsnbextension
  Using cached widgetsnbextension-3.5.1-py2.py3-none-any.whl (2.2 MB)
Collecting ipywidgets
  Using cached ipywidgets-7.5.1-py2.py3-none-any.whl (121 kB)
Collecting jupyter-core>=4.6.1
  Using cached jupyter_core-4.6.3-py2.py3-none-any.whl (83 kB)
Collecting jupyter-client>=5.3.4
  Using cached jupyter_client-6.1.6-py3-none-any.whl (108 kB)
Collecting nbformat
  Using cached nbformat-5.0.7-py3-none-any.whl (170 kB)
Collecting Send2Trash
  Using cached Send2Trash-1.5.0-py3-none-any.whl (12 kB)
Collecting argon2-cffi
  Using cached argon2_cffi-20.1.0-cp37-cp37m-win_amd64.whl (42 kB)
Collecting ipython-genutils
  Using cached ipython_genutils-0.2.0-py2.py3-none-any.whl (26 kB)
Collecting nbconvert
  Using cached nbconvert-5.6.1-py2.py3-none-any.whl (455 kB)
Collecting ipykernel
  Using cached ipykernel-5.3.4-py3-none-any.whl (120 kB)
Collecting traitlets>=4.2.1
  Using cached traitlets-4.3.3-py2.py3-none-any.whl (75 kB)
Collecting tornado>=5.0
  Using cached tornado-6.0.4-cp37-cp37m-win_amd64.whl (417 kB)
Collecting prometheus-client
  Using cached prometheus_client-0.8.0-py2.py3-none-any.whl (53 kB)
Collecting jinja2
  Using cached Jinja2-2.11.2-py2.py3-none-any.whl (125 kB)
Collecting pyzmq>=17
  Using cached pyzmq-19.0.2-cp37-cp37m-win_amd64.whl (1.0 MB)
Collecting terminado>=0.8.3
  Using cached terminado-0.8.3-py2.py3-none-any.whl (33 kB)
Collecting python-dateutil>=2.1
  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.2.0-cp37-none-win_amd64.whl (57 kB)
Collecting cycler>=0.10
  Using cached cycler-0.10.0-py2.py3-none-any.whl (6.5 kB)
Collecting pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting pillow>=6.2.0
  Using cached Pillow-7.2.0-cp37-cp37m-win_amd64.whl (2.1 MB)
Collecting ipython>=4.0.0; python_version >= "3.3"
  Using cached ipython-7.17.0-py3-none-any.whl (786 kB)
Collecting pywin32>=1.0; sys_platform == "win32"
  Using cached pywin32-228-cp37-cp37m-win_amd64.whl (9.1 MB)
Collecting jsonschema!=2.5.0,>=2.4
  Using cached jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting cffi>=1.0.0
  Using cached cffi-1.14.1-cp37-cp37m-win_amd64.whl (178 kB)
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting mistune<2,>=0.8.1
  Using cached mistune-0.8.4-py2.py3-none-any.whl (16 kB)
Collecting defusedxml
  Using cached defusedxml-0.6.0-py2.py3-none-any.whl (23 kB)
Collecting pygments
  Using cached Pygments-2.6.1-py3-none-any.whl (914 kB)
Processing c:\users\pc\appdata\local\pip\cache\wheels\63\99\01\9fe785b86d1e091a6b2a61e06ddb3d8eb1bc9acae5933d4740\pandocfilters-1.4.2-py3-none-any.whl
Collecting entrypoints>=0.2.2
  Using cached entrypoints-0.3-py2.py3-none-any.whl (11 kB)
Collecting bleach
  Using cached bleach-3.1.5-py2.py3-none-any.whl (151 kB)
Collecting testpath
  Using cached testpath-0.4.4-py2.py3-none-any.whl (163 kB)
Collecting decorator
  Using cached decorator-4.4.2-py2.py3-none-any.whl (9.2 kB)
Collecting MarkupSafe>=0.23
  Using cached MarkupSafe-1.1.1-cp37-cp37m-win_amd64.whl (16 kB)
Collecting pywinpty>=0.5; os_name == "nt"
  Using cached pywinpty-0.5.7-cp37-cp37m-win_amd64.whl (1.3 MB)
Collecting prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0
  Using cached prompt_toolkit-3.0.5-py3-none-any.whl (351 kB)
Collecting backcall
  Using cached backcall-0.2.0-py2.py3-none-any.whl (11 kB)
Collecting jedi>=0.10
  Using cached jedi-0.17.2-py2.py3-none-any.whl (1.4 MB)
Collecting pickleshare
  Using cached pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)
Requirement already satisfied: setuptools>=18.5 in c:\users\pc\bachelorthesis\venv\lib\site-packages (from ipython>=4.0.0; python_version >= "3.3"->ipywidgets->open3d) (46.4.0)
Collecting colorama; sys_platform == "win32"
  Using cached colorama-0.4.3-py2.py3-none-any.whl (15 kB)
Collecting attrs>=17.4.0
  Using cached attrs-19.3.0-py2.py3-none-any.whl (39 kB)
Processing c:\users\pc\appdata\local\pip\cache\wheels\22\52\11\f0920f95c23ed7d2d0b05f2b7b2f4509e87a20cfe8ea43d987\pyrsistent-0.16.0-cp37-cp37m-win_amd64.whl
Collecting importlib-metadata; python_version < "3.8"
  Using cached importlib_metadata-1.7.0-py2.py3-none-any.whl (31 kB)
Collecting pycparser
  Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
Collecting packaging
  Using cached packaging-20.4-py2.py3-none-any.whl (37 kB)
Collecting webencodings
  Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting wcwidth
  Using cached wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
Collecting parso<0.8.0,>=0.7.0
  Using cached parso-0.7.1-py2.py3-none-any.whl (109 kB)
Collecting zipp>=0.5
  Using cached zipp-3.1.0-py3-none-any.whl (4.9 kB)
Installing collected packages: pywin32, decorator, ipython-genutils, six, traitlets, jupyter-core, python-dateutil, tornado, pyzmq, jupyter-client, attrs, pyrsistent, zipp, importlib-metadata, jsonschema, nbformat, Send2Trash, pycparser, cffi, argon2-cffi, mistune, defusedxml, MarkupSafe, jinja2, pygments, pandocfilters, entrypoints, pyparsing, packaging, webencodings, bleach, testpath, nbconvert, wcwidth, prompt-toolkit, backcall, parso, jedi, pickleshare, colorama, ipython, ipykernel, prometheus-client, pywinpty, terminado, notebook, kiwisolver, numpy, cycler, pillow, matplotlib, widgetsnbextension, ipywidgets, open3d
Successfully installed MarkupSafe-1.1.1 Send2Trash-1.5.0 argon2-cffi-20.1.0 attrs-19.3.0 backcall-0.2.0 bleach-3.1.5 cffi-1.14.1 colorama-0.4.3 cycler-0.10.0 decorator-4.4.2 defusedxml-0.6.0 entrypoints-0.3 importlib-metadata-1.7.0 ipykernel-5.3.4 ipython-7.17.0 ipython-genutils-0.2.0 ipywidgets-7.5.1 jedi-0.17.2 jinja2-2.11.2 jsonschema-3.2.0 jupyter-client-6.1.6 jupyter-core-4.6.3 kiwisolver-1.2.0 matplotlib-3.3.0 mistune-0.8.4 nbconvert-5.6.1 nbformat-5.0.7 notebook-6.1.1 numpy-1.19.1 open3d-0.10.0.1 packaging-20.4 pandocfilters-1.4.2 parso-0.7.1 pickleshare-0.7.5 pillow-7.2.0 prometheus-client-0.8.0 prompt-toolkit-3.0.5 pycparser-2.20 pygments-2.6.1 pyparsing-2.4.7 pyrsistent-0.16.0 python-dateutil-2.8.1 pywin32-228 pywinpty-0.5.7 pyzmq-19.0.2 six-1.15.0 terminado-0.8.3 testpath-0.4.4 tornado-6.0.4 traitlets-4.3.3 wcwidth-0.2.5 webencodings-0.5.1 widgetsnbextension-3.5.1 zipp-3.1.0
WARNING: You are using pip version 20.2; however, version 20.2.1 is available.
You should consider upgrading via the 'c:\users\pc\bachelorthesis\venv\scripts\python.exe -m pip install --upgrade pip' command.

(venv) C:\Users\PC\Bachelorthesis>python -c "import open3d"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\PC\Bachelorthesis\venv\lib\site-packages\open3d\__init__.py", line 47, in <module>
    from open3d.open3d_pybind import camera
ImportError: DLL load failed: Das angegebene Modul wurde nicht gefunden.
```
I installed it on my notebook the same way and there it works fine and the `open3d_pybind.cp37-win_amd64.pyd` is in both directories I mentioned above. 

I also tried the steps you told @collikoira but I got this:
```
Microsoft Windows [Version 10.0.19041.388]
(c) 2020 Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\PC\Bachelorthesis\venv\Scripts>.\python.exe
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> quit()

C:\Users\PC\Bachelorthesis\venv\Scripts>.\python.exe -c "import open3d"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\PC\Bachelorthesis\venv\lib\site-packages\open3d\__init__.py", line 47, in <module>
    from open3d.open3d_pybind import camera
ImportError: DLL load failed: Das angegebene Modul wurde nicht gefunden.
```
But the python.exe und python37.dll is in the same directory:
![grafik](https://user-images.githubusercontent.com/56630682/89710992-c492f880-d987-11ea-86c7-264590c1ac44.png)

Now. I have noidea what I can do. Do you have one?
If you need more information, please let my know.
Thank you in advance.

**Comment by collikoira** [comment]
@yxlao Thanks, I found that I used 32 bit python version , after changing to 64bit module seems to be working, @Me817 can you make sure you are using 64bit python, Im using Python 3.7.7, Also using pycharm...

-Mayby action point for this issue is to check on installing which 64/32 bit version is user using.

**Comment by yxlao** [comment]
> -Mayby action point for this issue is to check on installing which 64/32 bit version is user using.
good point, thanks @collikoira

**Comment by Me817** [comment]
Hello, @yxlao and @collikoira thanks for the comments. 

I am using 64 bit version of python on my Computer and in PyCharm too. 
I think the problem might be that there is no `open3d_pybind.cp37-win_amd64.pyd` in the directories after installing open3d in the way I show in my last comment. What can I do?

Thanks in advance.

**Comment by Me817** [comment]
Hello, @yxlao and @collikoira thanks for the comments.

I am using 64 bit version of python on my Computer and in PyCharm too.
I tried to install open3d in PyCharm via the [Package Installer](https://www.jetbrains.com/help/pycharm/installing-uninstalling-and-upgrading-packages.html#install-package) and now the `open3d_pybind.cp37-win_amd64.pyd` is located in `C:\Users\PC\PycharmProjects\bachelorthesis\venv\Lib\site-packages\open3d`. So I used the "Dependencies" tool to locate `open3d_pybind.cp37-win_amd64.pyd`. This is the screenshot:
![Dependencies_screenshot](https://user-images.githubusercontent.com/56630682/89787504-db6b5380-db1d-11ea-8fc6-211ef53b6b50.png)

It seems that VCOMP140.DLL is not loaded.

**Comment by Me817** [comment]
Hello,

I solved my problem by downloading and installing [Visual C++ Redistributable fur Visual Studio 2015 from Official Microsoft Download Center](https://www.microsoft.com/de-DE/download/details.aspx?id=48145) according to the advice from this [forum](https://www.drwindows.de/programme-und-tools/165789-vcomp140-dll-gefunden.html). I installed vc_redist.x64.exe and everything works fine. 
Thanks to @yxlao to mention the Dependecies tool. It helped a lot discovering the missing .dll.
I think we can close this issue since all problems mentioned here are solved. Thanks for your help: @yxlao, @collikoira, @griegler.

**Comment by Me817** [comment]
Hello,

I solved my problem by downloading and installing Visual C++ Redistributable fur Visual Studio 2015 from Official Microsoft Download Center according to the advice from this forum. I installed vc_redist.x64.exe and everything works fine.
Thanks to @yxlao to mention the Dependecies tool. It helped a lot discovering the missing .dll.
I think we can close this issue since all problems mentioned here are solved. Thanks for your help: @yxlao, @collikoira, @griegler.
